tests/src/test/scala/cats/tests/TraverseSuite.scala

```diff
@@ -32,6 +32,12 @@ abstract class TraverseSuite[F[_]: Traverse](name: String)(implicit ArbFInt: Arb
     }
   }
 
+  test(s"Traverse[$name].traverseMatches_ with Option") {
+    forAll { (fa: F[Int], fn: Int => Option[Int]) =>
+      assert(Applicative[Option].void(fa.traverse(fn)) == fa.traverse_(fn))
```

Comment:
there is no law on this, and probably should be, but requires source and binary changes to laws.

Reply:
FYI laws can be introduced without breaking either source or binary compatibility